compat/inet_ntop.cc

```diff
@@ -163,24 +163,23 @@ inet_ntop6(const u_char *src, char *dst, size_t size)
      * Keep this in mind if you think this function should have been coded
      * to use pointer overlays.  All the world's not a VAX.
      */
-    char tmp[sizeof "ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255"], *tp;
```

Comment:
The same string literal is used below in the code (near line 228), and the two uses are dangerously tied to each other. Consider replacing that second use with `tmp`:

```diff
-        tp += snprintf(tp, (tmp + sizeof("ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255") - tp), "%x", words[i]);
+        tp += snprintf(tp, (tmp + sizeof(tmp) - tp), "%x", words[i]);
```

Reply:
I've factored out to a constexpr size variable. It may not be minimal but I think it better expresses intent